src/lib/verifyAppleIdToken.ts

```diff
@@ -1,51 +1,51 @@
-import * as jwt from 'jsonwebtoken';
-import * as jwksClient from 'jwks-rsa';
+import * as jwt from "jsonwebtoken";
+import * as jwksClient from "jwks-rsa";
+import { VerifyAppleIdTokenParams, VerifyAppleIdTokenResponse } from "./types";
 
-export const APPLE_BASE_URL = 'https://appleid.apple.com';
-export const JWKS_APPLE_URI = '/auth/keys';
+export const APPLE_BASE_URL = "https://appleid.apple.com";
+export const JWKS_APPLE_URI = "/auth/keys";
 
-export interface VerifyAppleIdTokenParams {
-  idToken: string;
-  clientId: string | string[];
-  nonce?: string;
-}
-
-export const getApplePublicKey = async (kid) => {
+export const getApplePublicKey = async (kid: string) => {
   const client = jwksClient({
     cache: true,
     jwksUri: `${APPLE_BASE_URL}${JWKS_APPLE_URI}`,
   });
-  const key: any = await new Promise((resolve, reject) => {
+  const key = await new Promise<jwksClient.SigningKey>((resolve, reject) => {
     client.getSigningKey(kid, (error, result) => {
       if (error) {
         return reject(error);
       }
       return resolve(result);
     });
   });
-  return (key.publicKey as jwksClient.CertSigningKey) || (key.rsaPublicKey as jwksClient.RsaSigningKey);
+  return key.getPublicKey();
 };
 
-export default async (params: VerifyAppleIdTokenParams) => {
+export const verifyToken = async (params: VerifyAppleIdTokenParams) => {
   const decoded = jwt.decode(params.idToken, { complete: true });
   const { kid, alg } = decoded.header;
+
   const applePublicKey = await getApplePublicKey(kid);
   const jwtClaims = jwt.verify(params.idToken, applePublicKey, {
-    algorithms: [alg],
+    algorithms: [alg as jwt.Algorithm],
     nonce: params.nonce,
-  });
-  if (!jwtClaims.iss || jwtClaims.iss !== APPLE_BASE_URL) {
-    throw new Error(`The iss does not match the Apple URL - iss: ${jwtClaims.iss} | expected: ${APPLE_BASE_URL}`);
+  }) as VerifyAppleIdTokenResponse;
+
+  if (jwtClaims?.iss !== APPLE_BASE_URL) {
+    throw new Error(
+      `The iss does not match the Apple URL - iss: ${jwtClaims.iss} | expected: ${APPLE_BASE_URL}`
+    );
   }
 
-  if (
-    (Array.isArray(params.clientId) && params.clientId.includes(jwtClaims.aud)) ||
-    jwtClaims.aud === params.clientId
-  ) {
+  const isFounded = []
+    .concat(jwtClaims.aud)
+    .some((_) => [].concat(params.clientId).includes(_));
```

Comment:
Can you please use `aud` instead of `_` or something like this? The `_` is generally known as unused "something" bet we are using it inside as an argument for `includes` function :pray: